File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        count = 0
        for i in s:
            while(count<len(t) and t[count]!= i):
                count += 1
            if count == len(t):
                return False
            count += 1

        return True
        

File: 0392-is-subsequence/0392-is-subsequence.java
public class Solution {
    public boolean isSubsequence(String s, String t) {
        int sLen = s.length(), tLen = t.length();
        int sIndex = 0, tIndex = 0;
        
        while (sIndex < sLen && tIndex < tLen) {
            if (s.charAt(sIndex) == t.charAt(tIndex)) {
                sIndex++;
            }
            tIndex++;
        }
        
        return sIndex == sLen;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isSubsequence("abc", "ahbgdc")); // Output: true
        System.out.println(solution.isSubsequence("axc", "ahbgdc")); // Output: false
    }
}
